local-macros/src/lib.rs

```diff
@@ -1389,3 +1393,75 @@ pub fn symbol_type(attr: TokenStream, item: TokenStream) -> TokenStream {
     }
     .into()
 }
+
+fn get_enum_unwrapped_match(argument: &Expr, ancestors: &[Ident]) -> TokenStream2 {
+    let mut unwrapped_variant_selector = quote! {
+        unwrapped
+    };
+    for (index, ancestor) in ancestors.iter().skip(1).rev().enumerate() {
+        let previous_ancestor = &ancestors[ancestors.len() - 2 - index];
+        unwrapped_variant_selector = quote! {
+            crate::#previous_ancestor::#ancestor(#unwrapped_variant_selector)
+        }
+    }
+
+    let last_variant = ancestors.last().unwrap();
+
+    quote! {
+        match #argument {
+            #unwrapped_variant_selector => unwrapped,
+            _ => panic!("Expected {}", stringify!(#last_variant)),
```

Comment:
I don't think I tested that this actually prints out the expected panic message (eg "Expected Expression")